### PR TITLE
Counting Sort in GO

### LINKED ISSUE #2145: Counting Sort
- [x] C
- [x] C++
- [x] C#
- [x] CoffeeScript
- [ ] Dart 
- [x] Go - @m-code12 
- [x] Java
- [x] JS
- [x] Kotlin - @sandhyabhan
- [x] Php
- [x] Python
- [x] Readme
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @shivamgupta57121

### PR DESCRIPTION
**Pull request by m-code12**
Fixes #2145 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Counting Sort in Go

#### Languages Used:
- Go

#### Files Added:
- Counting_Sort.go

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by m-code12** [comment]
@singh-shreya6 Please review and merge.

File: Counting_Sort/Counting_Sort.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
+package main
+import "fmt"
+
+const MaxUint = ^uint(0)
+const MaxInt = int(MaxUint >> 1)
+const MinInt = -MaxInt - 1
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:6]: Space after  -

File: Counting_Sort/Counting_Sort.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+/*
+
+Sample Input :
+    Enter the size of the array : 4                                                                                              
+    Enter the elements of the array : 12 29 10 78
+
+Sample Output :
+    After sorting [10 12 29 78]	
+
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:57]: This line is unnecessary

File: Counting_Sort/Counting_Sort.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        fmt.Scan(&a[i])
+    }
+    list := a    // A copy of the array 'a' is assigned to 'list' 
+    fmt.Println("before sorting", list)
+    countingSort(list)
+    fmt.Println("after sorting", list)
+}
+
+/*
+
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:50]: This blank line is unnecessary

File: Counting_Sort/Counting_Sort.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
+package main
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:1]: Please add one line problem description

**Comment by m-code12** [comment]
@singh-shreya6 I made the required changes. Please merge.

File: Counting_Sort/Counting_Sort.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    for _, v := range list {
+        if v > maxNumber {
+            maxNumber = v
+        }
+        if v < minNumber {
+            minNumber = v
+        }
+    }
+
+    count := make([]int, maxNumber-minNumber+1)
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:28]: Sorry, didn't notice it earlier. Please ad space around  - and  + here.

File: Counting_Sort/Counting_Sort.go [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    for _, x := range list {
+        count[x-minNumber]++
```
Comment (singh-shreya6) [inline Counting_Sort/Counting_Sort.go:31]: Here also, space around  -

**Comment by m-code12** [comment]
@singh-shreya6  Done the changes.

**Pull request merged**